tests/python/pants_test/backend/python/tasks/test_ctypes_integration.py

```diff
@@ -150,25 +187,45 @@ def test_ctypes_native_language_interop(self):
       pants_binary_strict_deps_failure = self.run_pants_with_workdir(
         command=['binary', self._binary_target_with_interop],
         # Explicitly set to True (although this is the default).
-        config={'native-build-settings': {'strict_deps': True}},
+        config={
+          # TODO(#6848): don't make it possible to forget to add the toolchain_variant option!
+          'native-build-settings': {
+            'toolchain_variant': toolchain_variant,
+            'strict_deps': True,
+          },
+        },
         workdir=os.path.join(buildroot.new_buildroot, '.pants.d'),
         build_root=buildroot.new_buildroot)
       self.assert_failure(pants_binary_strict_deps_failure)
-      self.assertIn("fatal error: 'some_math.h' file not found",
+      self.assertIn(self._include_not_found_message_for_variant[toolchain_variant],
                     pants_binary_strict_deps_failure.stdout_data)
 
-    pants_run_interop = self.run_pants(['-q', 'run', self._binary_target_with_interop], config={
-      'native-build-settings': {
-        'strict_deps': False,
-      },
+    # TODO(#6848): we need to provide the libstdc++.so.6.dylib which comes with gcc on osx in the
+    # DYLD_LIBRARY_PATH during the 'run' goal somehow.
+    attempt_pants_run = Platform.create().resolve_platform_specific({
+      'darwin': lambda: toolchain_variant != 'gnu',
+      'linux': lambda: True,
     })
-    self.assert_success(pants_run_interop)
-    self.assertEqual('x=3, f(x)=299\n', pants_run_interop.stdout_data)
+    if attempt_pants_run:
+      pants_run_interop = self.run_pants(['-q', 'run', self._binary_target_with_interop], config={
+        'native-build-settings': {
+          'toolchain_variant': toolchain_variant,
+          'strict_deps': False,
+        },
+      })
+      self.assert_success(pants_run_interop)
+      self.assertEqual('x=3, f(x)=299\n', pants_run_interop.stdout_data)
+
+  def test_ctypes_native_language_interop(self):
+    for variant in ToolchainVariant.allowed_values:
+      self._assert_ctypes_interop_with_mock_buildroot(variant)
 
   def test_ctypes_third_party_integration(self):
     pants_binary = self.run_pants(['binary', self._binary_target_with_third_party])
     self.assert_success(pants_binary)
 
+    # TODO(#6848): this fails when run with gcc on osx as it requires gcc's libstdc++.so.6.dylib to
+    # be available on the runtime library path.
     pants_run = self.run_pants(['-q', 'run', self._binary_target_with_third_party])
```

Comment:
Any way we could wrap this in a conditional that skips if the condition is true and logs this info to the user?

Reply:
I would prefer that this just gets fixed for real as part of #6848, as the underlying issue seems to require some infra work beforehand (figuring out how to run things which depend on runtime libraries from our `BinaryTool`s). Does that sound reasonable?